### ISSUE TITLE
Implementation with other starter kit

### ISSUE THREAD

**Issue by hundsim** [shakacode/react_on_rails#488]
Firstly, thanks for a very good and clever integration with Rails.

An idea though, why not piggyback on some of the established starter kits, e.g. https://github.com/coryhouse/react-slingshot/

Either fully implementing react-slingshot OR make the integration so thin that it would be easy to plug-and-play a starter kit of choice?

**Comment by justin808** [comment]
@hundsim What's to prevent you from using that? You just put what you need in the /client directory of your rails project, right?

Can you create a sample project and push to github so I can have a better idea of what you are suggesting?

**Comment by justin808** [comment]
@hundsim Any update?

**Comment by justin808** [comment]
@hundsim I think if you're thinking about the generator, we wouldn't want to do that as it would add a lot of complexity. CC: @jbhatab.

**Comment by jbhatab** [comment]
@hundsim I think we could talk to their team to see if they want to just have their starter kit use our gem, but our generator kinda does this already in terms of getting you up and running. 

I'll message them to see but I think just keeping our generator on our gem and their starter kit as their starter kit makes the most sense. 

Also we have an example app already built with react_on_rails. http://www.reactrails.com/

**Comment by hundsim** [comment]
I guess I was pondering about potential benefits of a slightly more modular structure of the gem, where one in a snap could swap in another react starter-kit - perhaps extracting the boilerplate code to another gem?

There's nothing wrong with your defaults, quite the contrary. But seeing how progressive [the react-landscape is](http://andrewhfarmer.com/starter-project/) - perhaps the maintenance of `react_on_rails` would be easier and more future-proof if the rails-integration and the react implementation wouldn't be so tightly bundled?


There is [such a diverse](http://andrewhfarmer.com/starter-project/) react-landscape regarding to starter-kits/boilerplate, and I think that speaks to the different approaches each and every developer has on how to 



I primarily mean the structure of folders/files and choice of modules etc, and only secondly care about the generator part where a test app is generated.


of various reasons I currently prefer some of the .

**Comment by hundsim** [comment]
I guess I was pondering about potential benefits of a slightly more modular structure of the gem, where one in a snap could swap in another react starter-kit - perhaps extracting the boilerplate code to another gem?

There's nothing wrong with your defaults, quite the contrary. But seeing how progressive [the react-landscape is](http://andrewhfarmer.com/starter-project/) - perhaps the maintenance of `react_on_rails` would be easier and more future-proof if the rails-integration and the react implementation wouldn't be so tightly bundled?

**Comment by jbhatab** [comment]
@hundsim Oh I get where you are going with it. Yeah we discussed just making this into just a webpack integration with no connection to the concept of react at all. 

We concluded that the initial default generator is so simple though that just deleting a few files abstracts it from react completely. Although having some options that instantly setup a variety of best practices is interesting, but would probably be hard to maintain in context to our generator. Maybe just a simple link to their start kit and saying look here for best practices would work?

I'm open to the idea of modularizing this gem if it's something you'd want to spearhead, but at this point I don't see it as something major enough worth pursuing.
